### PR TITLE
Enhancement/70 implement few shot pipeline

### PR DESCRIPTION
**Pull request by nils-schmitt**
This PR implements few-shot in every prompt to improve reliability of the model and to remove the need for function calling thus reducing tokan usage. Like stated 

"Hab jetzt auf meinem Branch soweit alle Prompts in few-shot umgewandelt. Habe in den Compares 80 und 78% erreicht und die log compares sind jetzt auf 5*(Länge(GroundTruth) + Länge(DF)) limitiert. Ebenfalls werden für die Timestamps jetzt nicht mehr immer der ganze Text sondern nur noch ein Ausschnitt bereitgestellt, wodurch die Token je Query nicht mehr von der Textlänge abhängig sind."

### CODE REVIEW

File: tracex/extraction/logic/modules/module_event_log_comparator.py [hunk 1]
```diff
@@ -49,37 +53,41 @@ def __start_comparison(self, pipeline_df, ground_truth_df):
...
-                "Percentage of event information in the ground truth that are contained event log by the pipeline: "
+                "Percentage of activities contained in ground truth that are found by the pipeline: "
                 + str(matching_perc_ground_truth_to_pipeline)
-                + "\n"
+                + "%\n"
             )
 
         return pipeline_df
 
     def __compare_activities(self, input_df, comparison_basis_df):
```
[the remark this answers is not in the document]
Comment (nils-schmitt) [reply tracex/extraction/logic/modules/module_event_log_comparator.py:68]: These are different things; 

`pipeline_df` and `gorund_truth_df` are the results, that we want to compare against each other. Contrary in `__compare_activities()` we want to compare any dataframes which is why they are named `input_df` and `comparison_basis_df` as we compare the `input` one to the `comparison_basis` as we check for every activity in the `input` one if it's also found in the `comparision_basis`. And as we use `__compare_activities()` once with the groundtruth first and once with groundtruth as comparison basis, we shouldn't name them similarly.

**Review by PitButtchereit — APPROVED** [review]

**Review by PitButtchereit — APPROVED** [review]

File: tracex/extraction/logic/modules/module_activity_labeler.py [hunk 2]
```diff
@@ -20,24 +20,24 @@ def __init__(self):
         self.description = "Extracts the activity labels from a patient journey."
 
     @log_execution_time(Path(settings.BASE_DIR / "tracex/logs/execution_time.log"))
-    def execute(self, _input, patient_journey=None):
-        super().execute(_input, patient_journey)
+    def execute(self, df, patient_journey=None, patient_journey_sentences=None):
```
Comment (tkv29) [inline tracex/extraction/logic/modules/module_activity_labeler.py:23]: replacement with df, please

File: tracex/extraction/logic/modules/module_event_log_comparator.py [hunk 3]
```diff
@@ -49,47 +53,44 @@ def __start_comparison(self, pipeline_df, ground_truth_df):
...
         total_matching_activities = 0
-        for activity in input_df["activity"]:
+        for i in range(len(input_df["activity"])):
             total_matching_activities += self.__find_activity(
```
Comment (tkv29) [inline tracex/extraction/logic/modules/module_event_log_comparator.py:71]: change to enumerate and rename variable i, please

File: tracex/extraction/logic/modules/module_event_log_comparator.py [hunk 4]
```diff
@@ -49,47 +53,44 @@ def __start_comparison(self, pipeline_df, ground_truth_df):
...
+
         return matching_percentage
 
     @staticmethod
-    def __find_activity(activity, comparison_basis_df):
-        for comparison_activity in comparison_basis_df["activity"]:
-            messages = [
-                {"role": "system", "content": p.COMPARE_CONTEXT},
+    def __find_activity(activity, comparison_basis_df, index):
+        lower = max(0, index - 2)
```
Comment (tkv29) [inline tracex/extraction/logic/modules/module_event_log_comparator.py:82]: comment or rename magic number

File: tracex/extraction/logic/modules/module_time_extractor.py [hunk 5]
```diff
@@ -21,68 +23,102 @@ def __init__(self):
...
+        try:
+            datetime.strptime(date_string, date_format)
+
+            return True
+        except ValueError:
+            return False
+
+    def __get_snippet(self, sentence_id):
+        """Extract the snippet for a given activity."""
+        lower = max(0, int(sentence_id) - 2)
```
Comment (tkv29) [inline tracex/extraction/logic/modules/module_time_extractor.py:120]: see comment above

**Review by tkv29 — COMMENTED** [review]
see comments above

File: tracex/extraction/logic/modules/module_time_extractor.py (hunk 5 above)
Comment (nils-schmitt) [reply tracex/extraction/logic/modules/module_time_extractor.py:123]: done

File: tracex/extraction/logic/modules/module_event_log_comparator.py (hunk 4 above)
Comment (nils-schmitt) [reply tracex/extraction/logic/modules/module_event_log_comparator.py:85]: done

File: tracex/extraction/logic/modules/module_event_log_comparator.py (hunk 3 above)
Comment (nils-schmitt) [reply tracex/extraction/logic/modules/module_event_log_comparator.py:71]: done

File: tracex/extraction/logic/modules/module_activity_labeler.py (hunk 2 above)
Comment (nils-schmitt) [reply tracex/extraction/logic/modules/module_activity_labeler.py]: done

**Review by tkv29 — APPROVED** [review]

**Pull request merged**